core/libcamera_app.cpp

```diff
@@ -122,7 +122,11 @@ void LibcameraApp::ConfigureViewfinder()
 	if (options_->verbose)
 		std::cout << "Configuring viewfinder..." << std::endl;
 
-	configuration_ = camera_->generateConfiguration({ StreamRole::Viewfinder });
+	bool have_lores_stream = options_->lores_width && options_->lores_height;
+	StreamRoles stream_roles = { StreamRole::Viewfinder };
+	if (have_lores_stream)
+		stream_roles.push_back(StreamRole::Viewfinder);
+	configuration_ = camera_->generateConfiguration(stream_roles);
 	if (!configuration_)
```

Comment:
Not sure I get this logic.  Why do a push_back on stream_roles conditionally?  Also, stream_roles will have 2 elements of StreamRole::Viewfinder.

Reply:
I guess I didn't want to force the creation of the low res stream if it wasn't needed. Unless it gets created anyway. But maybe that depends on whether colour denoise is running, or what size lo res stream we ask for...?? Maybe I'm a bit confused!

On the other point, I assume there's no problem in having two streams with the "viewfinder" role? I'm not too sure what other "role" I would give it!

Reply:
Makes sense, I did not think about it that way. I don't think there is any problem with having multiple Viewfinder roles in that case.